### PR TITLE
Unit test support in GAX

### PR DESCRIPTION
**Pull request by michaelbausor**
Add support for unit test in GAX
- Add MockStubTrait and MockUnaryCall
- Update existing tests

### CODE REVIEW

File: src/Testing/MockUnaryCall.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+namespace Google\GAX\Testing;
+
+
```
Comment (shinfan) [inline src/Testing/MockUnaryCall.php]: nit: empty line

File: src/Testing/MockUnaryCall.php [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            $status->setCode(Code::OK);
+        }
+        $this->status = $status;
+    }
+
+    public function wait()
+    {
+        return [call_user_func($this->deserialize, $this->response), $this->status];
+    }
+}
```
Comment (shinfan) [inline src/Testing/MockUnaryCall.php]: nit: newline EOF

File: src/Testing/MockUnaryCall.php [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+use google\rpc\Code;
+use google\rpc\Status;
+
+class MockUnaryCall
+{
+    private $response;
+    private $deserialize;
+    private $status;
+
+    public function __construct($response, $deserialize = null, $status = null)
```
Comment (shinfan) [inline src/Testing/MockUnaryCall.php]: Can you add comments to these public methods like you did in MockStub.php?

File: tests/ApiCallableTest.php [hunk 4]
```diff
@@ -165,7 +164,11 @@ public function testRetryTimeoutExceeds()
         $request = "request";
         $response = "response";
         $status = new MockStatus(Grpc\STATUS_DEADLINE_EXCEEDED, 'Deadline Exceeded');
-        $stub = MockStub::createWithResponseSequence([[$response, $status]]);
+        $stub = MockStub::createWithResponseSequence([
+            [$response, $status],
+            [$response, $status],
+            [$response, $status],
```
Comment (shinfan) [inline tests/ApiCallableTest.php]: nit: remove comma

File: src/Testing/MockStubTrait.php [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+namespace Google\GAX\Testing;
+
+trait MockStubTrait
```
Comment (shinfan) [inline src/Testing/MockStubTrait.php]: Add comments

File: src/Testing/MockStubTrait.php [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    public function addResponse($response, $status = null)
+    {
+        $this->responses[] = [$response->serialize(), $status];
+    }
+
+    public function getReceivedCalls()
+    {
+        $x = $this->receivedFuncCalls;
```
Comment (shinfan) [inline src/Testing/MockStubTrait.php]: Rename `$x` to be more descriptive.

File: src/Testing/MockUnaryCall.php (hunk 1 above)
Comment (michaelbausor) [inline src/Testing/MockUnaryCall.php]: Done

File: tests/ApiCallableTest.php (hunk 4 above)
Comment (michaelbausor) [inline tests/ApiCallableTest.php]: Done

File: src/Testing/MockUnaryCall.php (hunk 2 above)
Comment (michaelbausor) [inline src/Testing/MockUnaryCall.php]: Done

File: src/Testing/MockUnaryCall.php (hunk 3 above)
Comment (michaelbausor) [inline src/Testing/MockUnaryCall.php]: Done

File: src/Testing/MockStubTrait.php (hunk 6 above)
Comment (michaelbausor) [inline src/Testing/MockStubTrait.php]: Done

File: src/Testing/MockStubTrait.php (hunk 5 above)
Comment (michaelbausor) [inline src/Testing/MockStubTrait.php]: Done

**Pull request merged**